src/Microsoft.OData.Core/Json/JsonWriter.cs

```diff
@@ -621,6 +621,493 @@ public void EndTextWriterValueScope()
             }
         }
 
+        /// <inheritdoc/>
+        public Task StartPaddingFunctionScopeAsync()

```

Comment:
Do you think it would be easier to go through the file if we placed the async implementation in a separator file since this class is already declared partial? Feel free to ignore this if you think otherwise.